### PR TITLE
add test for partialSwitchUpdate

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 1]
```diff
@@ -333,4 +335,90 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+    @Tidy
+    @Tags(LOW_PRIORITY)
+    @Unroll
+    def "Able to partially update switch a 'location.#data.field' field"() {
+        given: "A switch"
+        def sw = topology.activeSwitches.first()
+        def initConf = northbound.getSwitch(sw.dpId)
+
+        when: "Request a switch partial update for a #data.field field"
+        def updateRequest = new SwitchPatchDto().tap {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy:348]: A more compact groovy way would be
```
def updateRequest =[location: [(data.field): data.newValue] as SwitchPatchDto
```

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 2]
```diff
@@ -333,4 +335,90 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+        }
+        def response = northboundV2.partialSwitchUpdate(sw.dpId, updateRequest)
+
+        then: "Update response reflects the changes"
+        response.location."$data.field" == data.newValue
+
+        and: "Changes actually took place"
+        northbound.getSwitch(sw.dpId).location."$data.field" == data.newValue
+
+        cleanup:
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy:359]: Just create a method that maps SwitchDto (initConf) to SwitchPatchDto and get rid of all these if-else branches

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy [hunk 3]
```diff
@@ -333,4 +335,90 @@ class SwitchesSpec extends HealthCheckSpecification {
...
+        def response = northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap { it.pop = newPopValue })
+
+        then: "Update response reflects the changes"
+        response.pop == newPopValue
+
+        and: "Changes actually took place"
+        northbound.getSwitch(sw.dpId).pop == newPopValue
+
+        cleanup:
+        if (initConf.pop) {
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy:418]: northboundV2.partialSwitchUpdate(sw.dpId, new SwitchPatchDto().tap { it.pop = initConf.pop ?: "" }

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesSpec.groovy:357]: I didn't extract it into separate method(I will do it when we start use this kind of operation more often).
For now I just refactored this block of code

**Pull request merged**